sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -26,15 +26,18 @@
      * @return An object with defaults filled in for null values in the original.
      */
     public static ParallelTransferOptions populateAndApplyDefaults(ParallelTransferOptions other) {
-        other = other == null ? new ParallelTransferOptions(null, null, null) : other;
-        return new ParallelTransferOptions(
-            other.getBlockSize() == null ? Integer.valueOf(BlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE)
-                : other.getBlockSize(),
-            other.getNumBuffers() == null ? Integer.valueOf(BlobAsyncClient.BLOB_DEFAULT_NUMBER_OF_BUFFERS)
-                : other.getNumBuffers(),
-            other.getProgressReceiver(),
-            other.getMaxSingleUploadSize() == null ? Integer.valueOf(BlockBlobAsyncClient.MAX_UPLOAD_BLOB_BYTES)
-                : other.getMaxSingleUploadSize());
+        other = other == null ? new ParallelTransferOptions() : other;
+        return new ParallelTransferOptions()
+            .setBlockSize(other.getBlockSizeLong() == null
+                ? Long.valueOf(BlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE)
```

Comment:
Shouldn't need `Long.valueOf`

Reply:
I did play with `valueOf` in ModelHelpers. After removing I got quite legit bug report.
![image](https://user-images.githubusercontent.com/61715331/78587449-d013be80-77f1-11ea-95ab-7363901ece3d.png)
So I ended up reworking this code little bit.